src/jimgw/core/jim.py

```
@@ -1,498 +1,528 @@
+        else:
+            logger.debug(
+                f"Using {len(likelihood_transforms)} likelihood transform(s): {[type(t).__name__ for t in likelihood_transforms]}"
+            )
 
-        Applies sample transforms in reverse to map from sampling space to prior
-        space before evaluating the prior, accumulating log-Jacobian corrections.
+        # Build sampling-space callables. These operate on flat arrays of shape
+        # (n_dims,) and are injected into the sampler.
+        names = self.parameter_names
+
+        def _log_prior_fn(arr: Float[Array, " n_dims"]) -> Float:
```

Ruff (0.15.15)

[error] Syntax error in forward annotation: Expected an expression

(F722)